overlord/snapshotstate/snapshotmgr.go

```diff
@@ -286,10 +286,14 @@ func doRestore(task *state.Task, tomb *tomb.Tomb) error {
 		return err
 	}
 
-	buf, err := json.Marshal(reader.Conf)
```

Comment:
I see code like this also in undoRestore, maybe there is more?

Reply:
looking at the code maybe we need to introduce a couple of helpers to avoid this repetition of delicate code

Reply:
As far as snapshots are concerned it's only restore/undorRestore (both covered now in this PR). The problem is specific to snapshots I think. For regular config ops we already have very explicit handling of nils for their special semantics.